csrc/scheduler/normalization_inner_tma.cpp

```diff
@@ -5,62 +5,296 @@
-      "Incorrect parameters sent to inner persistent TMA scheduler");
-
-  // TODO: Implement TMA-specific scheduling
-  // For now, this is a placeholder
-  normalization_scheduler_utils::schedulePersistentKernel(
-      fusion, rparams, rparams->scheduler_type);
+
+  // Use the reduction tensor as the starting point for scheduling
+  // Its transformations will be propagated to all non-TMA tensors
+  const auto& reduction_tvs = scheduler_utils::getReductionTvs(fusion);
+  TensorView* reduction_tv = reduction_tvs.at(0);
+
+  const scheduler_utils::PersistentBufferInfo persistent_info =
+      scheduler_utils::persistentBuffers(fusion);
+  std::vector<TensorView*> dummy_outputs;
+  if (params->project_persistent_buffers) {
+    dummy_outputs = reduction_scheduler_utils::projectPersistentBuffers(
+        fusion, persistent_info, params->project_persistent_buffers);
+    for (auto output : dummy_outputs) {
+      fusion->addOutput(output);
+    }
+  } else {
+    NVF_ERROR(
+        false, "Non-projectable buffers are not supported in TMA version yet");
+  }
+  // Identify persistent inputs that will be cached in shared memory using
+  // TMA load (CpAsyncBulk) for efficient async memory transfers
+  std::unordered_set<TensorView*> persistent_inputs{
+      persistent_info.projectable_buffer_inputs.begin(),
+      persistent_info.projectable_buffer_inputs.end()};
+  for (auto buffer : persistent_info.persistent_buffers) {
+    if (buffer->isFusionInput()) {
+      persistent_inputs.insert(buffer);
+    }
+  }
+  // Categorize cached inputs into TMA loads and regular LDG loads
+  // - tma_tvs: Persistent inputs loaded via TMA (CpAsyncBulk) to shared memory
+  // - ldg_tvs: Non-persistent inputs loaded via regular LDG instructions
+  // - smem2reg_tvs: Intermediate register caches for vectorized smem->reg loads
+  std::vector<TensorView*> ldg_tvs, tma_tvs, smem2reg_tvs;
+  const auto& cached_inputs = scheduler_utils::cacheInputs(fusion, true);
+  for (auto [tv, input_idx] : cached_inputs) {
+    auto input = fusion->inputs()[input_idx]->as<TensorView>();
+    if (!params->tma_load_non_persistent_buffers &&
+        !persistent_inputs.contains(input)) {
+      // Non-persistent input: use regular global load
+      ldg_tvs.push_back(tv);
+      continue;
+    }
+    if (auto load_op = dynamic_cast<LoadStoreOp*>(tv->definition())) {
+      // Persistent input: use TMA load to shared memory
+      load_op->setOpType(LoadStoreOpType::CpAsyncBulk);
+      tv->setMemoryType(MemoryType::Shared);
+      tma_tvs.push_back(tv);
+      if (params->vectorize_load_smem_to_regs) {
+        // Create register cache for vectorized smem->reg loads
+        auto regs_cache = tv->cacheAfter();
+        smem2reg_tvs.push_back(regs_cache);
+        // Replicate cache for multiple consumers to avoid conflicts
+        const auto& consumers = ir_utils::consumerTvsOf(regs_cache);
+        for (auto i = 1; i < (int)consumers.size(); i++) {
```

Comment:
Drop first element with `std::views`
```suggestion
        for (auto consumer : ir_utils::consumerTvsOf(regs_cache) | std::views::drop(1)) {
```